src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,156 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String OUT_OF_BOUNDS_EXCEPTION = "Index out of bounds!";
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> nodeToAdd = new Node<>(null, value, null);
+        if (size == 0) {
+            first = nodeToAdd;
+        } else {
+            nodeToAdd.prev = last;
+            nodeToAdd.prev.next = nodeToAdd;
+        }
+        last = nodeToAdd;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        rangeCheckForAdd(index);
+        Node<T> nodeToAdd = new Node<>(null, value, null);
+        Node<T> nodeToShift = nodeOf(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            first.prev = nodeToAdd;
+            nodeToAdd.next = first;
+            first = nodeToAdd;
+        }
+        nodeToShift.prev.next = nodeToAdd;
+        nodeToAdd.prev = nodeToShift.prev;
+        nodeToAdd.next = nodeToShift;
+        nodeToShift.prev = nodeToAdd;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
```

Comment:
Let's use `checkIndex` in `nodeOf()` method. This will reduce the number of reused code.

Reply:
Smart